### PR TITLE
core/txbuilder: add explicit retire action

### PR DESCRIPTION
**Pull request by jbowens**
Add an explicit retire action instead of using the `control_program`
action and specifying the `OP_FAIL` program. This provides more
flexibility as our implementation of retirement changes.

### CODE REVIEW

**Comment by jbowens** [comment]
PTAL @jeffomatic

File: sdk/java/src/main/java/com/chain/api/Transaction.java [hunk 1]
```diff
@@ -927,8 +927,7 @@ public ControlWithProgram setAmount(long amount) {
        * Default constructor defines the action type as "control_program"
        */
       public Retire() {
-        this.put("type", "control_program");
-        this.put("control_program", ControlProgram.retireProgram());
+        this.put("type", "retire");
```
Comment (jeffomatic) [inline sdk/java/src/main/java/com/chain/api/Transaction.java]: We should delete `ControlProgram.retireProgram()` and related dependencies.

Comment (jbowens) [inline sdk/java/src/main/java/com/chain/api/Transaction.java]: I think I did?

Comment (jeffomatic) [inline sdk/java/src/main/java/com/chain/api/Transaction.java]: Yep sorry I missed it 😞

**Pull request closed without merging**